gui/create-item.lua

```diff
@@ -184,11 +211,176 @@ local function getCreatureRaceAndCaste(caste)
  return df.global.world.raws.creatures.list_creature[caste.index],df.global.world.raws.creatures.list_caste[caste.index]
 end
 
+function createCorpsePiece(creator, bodypart, partlayer, creatureID, casteID, generic, quality) -- this part was written by four rabbits in a trenchcoat (ppaawwll)
+ -- (partlayer is also used to determine the material if we're spawning a "generic" body part (i'm just lazy lol))
+ quality = math.max(0, math.min(5, quality - 1))
+ creatureID = tonumber(creatureID)
+ -- get the actual raws of the target creature
+ local creatorRaceRaw = df.creature_raw.find(creatureID)
+ casteID = tonumber(casteID)
+ bodypart = tonumber(bodypart)
+ partlayer = tonumber(partlayer)
+ -- get body info for easy reference
+ local creatorBody = creatorRaceRaw.caste[casteID].body_info
+ 
+ local layerName
+ local layerMat
+ local tissueID
+ 
+ if not generic then -- if we have a specified body part and layer, figure all the stuff out about that
+ -- store the tissue id of the specific layer we selected
+  tissueID = tonumber(creatorBody.body_parts[bodypart].layers[partlayer].tissue_id)
+  layerMat = creatorRaceRaw.tissue[tissueID].tissue_material_str[1]
+  layerName = creatorBody.body_parts[bodypart].layers[partlayer].layer_name
+ else -- otherwise, figure out the mat name from the dual-use partlayer argument
+  layerMat = creatorRaceRaw.material[partlayer].id
+  layerName = layerMat
+ end
+
+ -- default is MEAT, so if anything else fails to change it to something else, we know that the body layer is a meat item
+ local item_type = "MEAT"
+ -- get race name and layer name, both for finding the item material, and the latter for determining the corpsepiece flags to set
+ local raceName = string.upper(creatorRaceRaw.creature_id)
+ -- every key is a valid non-hair corpsepiece, so if we try to index a key that's not on the table, we don't have a non-hair corpsepiece
+ local corpseTable = {BONE = "BONE", SKIN = "SKIN", CARTILAGE = "CARTILAGE", TOOTH = "TOOTH", NERVE = "NERVE", NAIL = "NAIL", HORN = "HORN", HOOF = "HOOF"}
+ -- we do the same as above but with hair
+ local hairTable = {HAIR = "HAIR", EYEBROW = "EYEBROW", EYELASH = "EYELASH", MOUSTACHE = "MOUSTACHE", CHIN_WHISKERS = "CHIN_WHISKERS", SIDEBURNS = "SIDEBURNS"}
```

Comment:
for sets that you intend to just test for membership, a better idiom is to put these outside of the function (since they're constants) and initialize them like this:
```lua
local CORPSE_PIECES = utils.invert{'BONE', 'SKIN', 'CARTILAGE', 'TOOTH', 'NERVE', 'NAIL', 'HORN', 'HOOF' }
```

Reply:
trueeeeeeeeeee.

Reply:
> thank you for fixing build-now : ) It would make things easier to review and merge if you were to create branches for separate changes and open separate PRs -- committing to your own `master` branch will give you merge issues further down the line. The usual flow is:
> 
> 1. sync your `master` branch from upstream
> 2. create a new branch on your fork from your updated `master`
> 3. commit to that fork
> 4. open a PR from that fork to DFHack/scripts
> 5. PR gets merged
> 6. delete the feature branch from your fork (there's a button for this on the PR after it gets merged)

ahhhhhh, thank you, makes sense. i'll make sure to do this in the future.